pkg/ipcache/ipcache.go

```diff
@@ -104,6 +103,7 @@ type K8sMetadata struct {
 // Configuration is init-time configuration for the IPCache.
 type Configuration struct {
 	context.Context
+	Logger *slog.Logger
```

Comment:
all other things here are embedded, could embed the logger too? would avoid adding the duplicate field on the ipc struct too